lib/helpers/truncateByBytes.js

```diff
@@ -0,0 +1,21 @@
+export const byteCount = (string) => encodeURI(string).split(/%..|./).length - 1;
+
+export const truncateByBytes = (string, byteSize) => {
+  if (byteCount(string) > byteSize) {
+    const charsArray = string.split('');
+    const truncatedStringArray = [];
+    let bytesCounter = 0;
+    for (let i = 0; i < charsArray.length; i += 1) {
+      bytesCounter += byteCount(charsArray[i]);
+      if (bytesCounter <= byteSize) {
+        truncatedStringArray.push(charsArray[i]);
```

Comment:
Instead of building an array of strings, can we instead maintain the index? Then we would just return `string.substring(0, bytesIndex)` or something.

So if index 0 is 2 bytes, `bytesCounter` would still be 2, but `bytesIndex` would be 1 after reading that first char. If `maxBytes` is 2, then we'd return `string.substring(0, bytesIndex)` which would be `string.substring(0, 1)`.

Does that make sense?